src/kernel-install/kernel-install.in

```diff
@@ -159,7 +159,7 @@ if [ -z "$MACHINE_ID" ] && [ -f /etc/machine-info ]; then
         log_verbose "machine-id $MACHINE_ID acquired from /etc/machine-info"
 fi
 if [ -z "$MACHINE_ID" ] && [ -f /etc/machine-id ]; then
-    read -r MACHINE_ID </etc/machine-id
+    read -r MACHINE_ID </etc/machine-id || true
```

Comment:
Maybe change `-f` to `-s` above? That way we don't lose other errors (like permission denied and so on)